persistent/test/Database/Persist/THSpec.hs

```diff
@@ -24,7 +24,7 @@
 module Database.Persist.THSpec where
 
 import Control.Applicative (Const(..))
-import Data.Aeson hiding (Key)
+import Data.Aeson (decode, encode)
```

Comment:
Surfaced problem in tests by doing this

Reply:
Thanks for resolving this! Do you think it is worth adding a specific spec to cover imports required by users of persistent? Something like a test that only imports persistent, and defines some entities (some with derivied json instances) or do you think that might be overkill, something like:

```
module Database.Persist.TH.JsonEncodingSpec where

-- this test asserts this is the only import required
import Database.Persist.TH

mkPersist sqlSettings [persistLowerCase|
JsonEncoding json
    name Text
    age  Int
    Primary name
    deriving Show Eq
JsonEncoding2 json
    name Text
    age Int
    blood Text
    Primary name blood
    deriving Show Eq
JsonEncMigrationOnly json
    name Text
    age  Int
    foo  Text MigrationOnly
|]
```

My original PR added to `JsonEncodingSpec` but that was already importing `Data.Aeson` :sweat_smile: what you have done here seems to be a more robust solution, but I wonder if we should also add a test similar to the above? Happy to do the PR for this if so :smile:

Reply:
Yeah, that's a good idea.